src/main/java/wolox/gmjavatraining/GmJavaTrainingApplication.java

```diff
@@ -2,7 +2,11 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
+@EnableJpaRepositories("wolox.gmjavatraining.persistence.repo")
```

Comment:
Remove both of these annotations, they're only needed when your files that need to be scanned are outside the default project structure (i.e. outside src/main/java).
It will break your autowired injections like this.